psycopg/psycopg/connection.py

```diff
@@ -125,6 +147,9 @@ def __init__(self, pgconn: "PGconn"):
         # apart a connection in the pool too (when _pool = None)
         self._pool: Optional["BasePool[Any]"]
 
+        # Attribute set when the connection enters pipeline mode.
+        self._pipeline_mode = False
```

Comment:
Why do you say that the code would be more complicated using `PGconn.pipeline_status()`? It doesn't seem to create problems in async too. I would prefer to do that: where we could, I tried to avoid to add Python state and rely on the pgconn state only (e.g. there are no explicit status, transaction_status, closed...)